cloud_storage/cloud_storage/overrides/file.py

```diff
@@ -50,17 +52,74 @@ def on_trash(self) -> None:
 				_("This file is attached to a submitted document and cannot be deleted"),
 				frappe.PermissionError,
 			)
-
+		if len(self.file_association) > 1:
+			frappe.throw(
+				_("This file is attached to multiple documents and cannot be deleted"),
+				frappe.PermissionError,
+			)
 		super().on_trash()
 
+	def associate_files(self) -> File:
```

Comment:
I don't think this is correct. The function doesn't seem to be returning anything.

Reply:
It used to, but it didn't have the desired effect and I didn't remove it.